transmogrifier/models.py

```diff
@@ -49,25 +43,42 @@ def not_empty(
         raise ValueError(message)
 
 
-@define
-class ListField:
-    def __hash__(self) -> int:
-        """Hash method to create unique identifier for Location objects."""
-        values = tuple(
-            [
-                tuple(attrib) if isinstance(attrib, list) else attrib
-                for attrib in attrs.astuple(self)
-            ]
-        )
-        return hash(values)
+def timdex_object_hash(timdex_object: Any) -> int:  # noqa: ANN401
+    """Hash method for TIMDEX objects.
+
+    This method is set as the hash method for TIMDEX objects.
+    The method generates a unique hash using a tuple
+    comprised of the class name and attribute values.
+    By making TIMDEX objects hashable, dedupe methods
+    can be applied to a list of TIMDEX objects.
+    """
+    values = tuple(type(timdex_object).__name__)
```

Comment:
A couple requests here.  

**First**, I think this approach might actually creating a tuple with the individual characters of the class name, e.g.

```python
from transmogrifier.models import *

timdex_object = AlternateTitle(value='horse',kind='animal')

tuple(type(timdex_object).__name__)
# Out[13]: ('A', 'l', 't', 'e', 'r', 'n', 'a', 't', 'e', 'T', 'i', 't', 'l', 'e')
```

Think this could be resolved via an explicit list, and then converting to tuple:
```python
tuple([type(timdex_object).__name__])
# Out[2]: ('AlternateTitle',)
```

**Second**, any objection with using `X.__class__` vs `type(X)`?

Example:
```python
values = tuple([timdex_object.__class__.__name__])
```

Personally, when we _know_ it's some kind of well-known class instance (like classes we define), I think using `.__class__` is more readable.  I'm having trouble finding supporting documentation or reasoning, where it sounds like `type(X)` and `X.__class__` are mostly interchangeable; [this SO post](https://stackoverflow.com/questions/9610993/python-type-or-class-or-is) being a good example where it's kind of fluid, matter of taste, and some pretty edge-y gotchas.

I tend to think of `type(X)` as when I really don't know anything about an object. But in this case, where we _know_ it's a class instance of some kind, and _we_ wrote the classes, it feels like accessing it directly is kind of nice.  

**TL/DR**: I'd propose we make the first change to avoid a tuple of each character, but the second `type(X)` vs `X.__class__` happy either way.

Reply:
ALL that said -- and should have led with this -- this is a nice and elegant solution to this equality checking!  Nice work.

Reply:
I consider `values = tuple([timdex_object.__class__.__name__])` more readable as well, good suggestion!

Reply:
EGAD 😱

Reply:
@ghukill Thank you for catching this erroneous error 🙈 . This has been fixed by https://github.com/MITLibraries/transmogrifier/pull/206/commits/426b185b94ab7194345c1c2a23dbb57b2fa15eb6. I also prefer using `__class__` for the same reason you described: 
> we _know_ it's a class instance of some kind, and _we_ wrote the classes

(Also remembering you suggested this during our check-in yesterday so apologies for forgetting!)